falcon/media/multipart.py

```diff
@@ -0,0 +1,412 @@
+
+    Note:
+        `BodyPart` is meant to be instantiated directly only by the
+        `MultipartForm` parser.
+
+    Attributes:
+        content_type (str): Value of the Content-Type header, or the multipart
+            form default ``text/plain`` if the header is missing.
+
+        data (bytes): Body part content bytes. The maximum number of bytes that
+            could be read is configurable via :class:`MultipartParseOptions`,
+            and a :class:`.MultipartParseError` is raised if the body part is
+            larger that this size.
+
+            For large bodies, such as attached files, use the input `stream`
+            directly.
+
+            .. note::
+               Accessing this property the first time would consume the part
+               input stream.
+               The value is cached for subsequent access.
+
+        filename (str): File name if the body part is an attached file, and
+            ``None`` otherwise.
+
+        secure_filename (str): The sanitized version of `filename` using only
+            the most common ASCII characters for maximum portability and safety
+            wrt using this name as a filename on a regular file system.
+
+        stream: File-like input object for reading the body part of the
+            multipart form request, if any. This object provides direct access
+            to the server's data stream and is non-seekable. The stream is
+            automatically delimited according to the multipart stream boundary.
+
+        media (object): Returns a deserialized form of the multipart body part.
+            When called, it will attempt to deserialize the body part stream
+            using the Content-Type header as well as the media-type handlers
+            configured via :class:`MultipartParseOptions`.
+
+
+        text (str): The part decoded as a text string provided the part is
+            encoded as ``text/plain``, ``None`` otherwise.
+
+            .. note::
+               As this property builds upon `data`, it would consume the part
+               input stream in the same way.
+    """
+
+    _content_disposition = None
+    _data = None
+    _filename = None
+    _media = None
+    _name = None
+
+    def __init__(self, stream, headers, parse_options):
+        self.stream = stream
+        self._headers = headers
+        self._parse_options = parse_options
+
+    @property
+    def data(self):
+        if self._data is None:
+            max_size = self._parse_options.max_body_part_buffer_size + 1
+            self._data = self.stream.read(max_size)
+            if len(self._data) >= max_size:
+                raise MultipartParseError('body part is too large')
+
+        return self._data
+
+    @property
+    def text(self):
+        content_type, options = cgi.parse_header(self.content_type)
+        if content_type != 'text/plain':
+            return None
+
+        charset = options.get('charset') or self._parse_options.default_charset
```

Comment:
Any particular reason not to just do the following? 

```py
options.get('charset', self._parse_options.default_charset)
```

Reply:
No, I just probably wrote that without thinking too much. We do not actually need to be too forgiving here.